front/app/components/Edits/ExpandedEdit/WikiExpandedEdit.tsx

```diff
@@ -0,0 +1,146 @@
+import * as React from 'react';
+import * as FontAwesome from 'react-fontawesome';
+import { Row, Col } from 'react-bootstrap';
+
+import { WikiPageEditFragment } from 'types/WikiPageEditFragment';
+
+interface EditProps {
+  edit: WikiPageEditFragment;
+}
+
+interface EditState {
+  lineVisible: any;
+  hoveredLine: number | null;
+}
+
+class WikiExpandedEdit extends React.Component<EditProps, EditState> {
+  state = {
+    lineVisible: {},
+    hoveredLine: null,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      lineVisible: this.prepareSpans(),
+      hoveredLine: null,
+    };
+  }
+
+  /**
+   * Only show a line on either side for context
+   * by default for line-by-line change.
+   */
+  prepareSpans = () => {
+    const { edit } = this.props;
+    let lastWasChange = false;
+    const lineVisible = {};
+    edit.changeSet.editLines.forEach((line, i) => {
+      if (line.status !== 'UNCHANGED') {
+        lineVisible[i - 1] = true;
+        lineVisible[i] = true;
+        lastWasChange = true;
+      } else if (lastWasChange) {
+        lineVisible[i] = true;
+        lastWasChange = false;
+      } else {
+        lineVisible[i] = false;
+      }
+    });
+    return lineVisible;
+  };
+
+  expandSpans = (start, end) => () => {
+    const lineVisibleUpdate = {};
+    for (let i = start; i <= end; i++) {
+      lineVisibleUpdate[i] = true;
+    }
+    this.setState({
+      lineVisible: { ...this.state.lineVisible, ...lineVisibleUpdate },
+    });
+  };
+
+  hoveredOpts = (key, className = '') => ({
+    key,
+    className: `${className}${
+      this.state.hoveredLine === key ? ' hovered' : ''
+    }`,
+    onMouseEnter: () => {
+      console.log(key);
+      this.setState({ ...this.state, hoveredLine: key });
+    },
+    onMouseLeave: () => this.setState({ ...this.state, hoveredLine: null }),
+  });
+
+  render() {
+    const {
+      edit: {
+        changeSet: { editLines },
+      },
+    } = this.props;
+    const { lineVisible, hoveredLine } = this.state;
+    const nodes: any[] = [];
+    const actions: any[] = [];
+    let firstInvisible: number | null = null;
+    editLines.forEach((line, i) => {
```

Comment:
In this section, we serially evaluate each change to determine whether the line has been expanded, creating an action for all groupings of unexpanded lines. By default we show the previous and subsequent line of a set of changes.